docs/Middleware-Guide.md

```diff
@@ -0,0 +1,82 @@
+Some examples of on-chain-checkable, objectively attributable behavior: double-signing a block in Ethereum, but NOT inactivity leak; proofs-of-custody in EigenDA (or Ethereum after Danksharding is activated), but NOT a node ceasing to serve data; a node in a light-node-bridge middleware signing an invalid block from another chain, but NOT the node failing to sign for a brief period.
+
+4. *Single Point-of-Interaction for Services and EigenLayer*: It is assumed that services have a single contract that coordinates the service’s communications sent to EigenLayer. This contract – referred to as the ServiceManager – informs EigenLayer of operator registration, updates, and deregistration, as well as signaling to EigenLayer when an operator should be slashed (frozen). Middlewares have full control over how they split up the actual logic involved, but are expected to route all calls to EigenLayer through a single contract. While technically possible, middlewares SHOULD NOT use multiple contracts to interact with EigenLayer, as this will impose additional burden on stakers and operators in EigenLayer when they are withdrawing or deregistering.
+
+## Integration with EigenLayer Contracts:
+In this section, we will explain various API interfaces that EigenLayer provides which are essential for middlewares to integrate with EigenLayer. 
+
+### *Opting into middleware*
+In order for any EigenLayer operator to be able to opt-in to a middleware, EigenLayer provides two interfaces: `optIntoSlashing(..)` and `recordFirstStakeUpdate(..)`. The sequential flow for opting into a middleware using these functions is as follows:
+1. The operator first opts into slashing by calling  `optIntoSlashing(..)` in `Slasher.sol`, where it has to specify the address of the middleware's ServiceManager contract in the argument. This step results in the operator giving permission to the middleware's ServiceManager contract to slash the operator via EigenLayer, if the operator is ever proven to have engaged in adversarial behavior while responding to the middleware's task. Successful call to  `optIntoSlashing(..)` emits the `OptedIntoSlashing(..)` event.
+2. Next, the operator needs to register with the middleware on chain via a middleware-specific registry contract (see the *Guide To Provided Middleware Contracts* section for examples). To integrate with EigenLayer, the middleware's Registry contract provides a registration endpoint that calls on the `recordFirstStakeUpdate(..)` in the middleware's ServiceManager contract which in turn calls the `recordFirstStakeUpdate(..)` function in `Slasher.sol`. On successful execution of this function call, the event `MiddlewareTimesAdded(..)` is emitted and the operator has to start serving the tasks from the middleware.
```

Comment:
"Guide To Provided Middleware Contracts" should be a link to that section

Reply:
typo: "calls on" => "calls"

Reply:
also suggest revising "has to" to something more neutral, like "is expected to" or "may"